common/priorityQueue.py

```diff
@@ -0,0 +1,81 @@
+import bisect
+
+class PriorityQueue:
+    def __init__(self):
+        self._queue = []
+        self._size = 0
+        self._id = 0
+
+        # a unique sender is determined by the path to the sender's emails
+        self._uniqueSenders = set([])
+
+    def getLength(self):
+        return self._size
+
+
+    # adds item to priority queue if among the top 10 emails with highest priority
+    # only one email is maintained in the priority queue per sender
+    def push(self, item, priority):
+        # checks to see if this sender is already in that priority queue
+        # if so, checks to see if probability should be updated
+        thisSender = item[0]
+        if thisSender in self._uniqueSenders:
+            senderIndex = None
+            for i in range(len(self._queue)):
+                if self._queue[i][2][0] == thisSender:
```

Comment:
self._queue has a tuple with three values: (priority, self._id, and item). The priority is the probability and the item is an email represented by a numpy array. I will add a comment with this info.